### DISCUSSION TITLE
请问table组件如何设置列宽跟着本列最长内容自动调整，不要隐藏每行的内容，要单行模式，把内容全显示出来

### DISCUSSION THREAD

**Discussion by dikongpulu** [layui/layui#1836] [问答]
就像excel这样
![image](https://github.com/layui/layui/assets/52130261/073bfb60-35b9-4592-84e8-8ed8dd3a0edd)

**Comment by Sight-wcg** [comment]
目前不支持根据单元格内容设置列宽

**Reply by dikongpulu** [reply]
这不应该是table原生就有的功能吗，能否通过另外设置样式达到效果呢

**Reply by Sight-wcg** [reply]
理论上只通过设置样式没法实现，但你可以试一试。

**Comment by LSL1618** [comment]
不知道字符个数与列宽度的转换比例是多少，而且字符又有很多种，中文、英文(大小写)、各种符号、特殊字符、emoji表情等等，想法是好的，但是估计最多只能预估一个大概的宽度。Excel中也只是预估宽度，并不准确。
